### DISCUSSION TITLE
2007/04/14/g-%E8%AF%AD%E8%A8%80

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#151] [Announcements]
# 2007/04/14/g-%E8%AF%AD%E8%A8%80

G 语言是图形化编程语言（Graphical Programing Language）的缩写。LabVIEW 有的时候也被叫做 G 语言。我们可以这样理解：LabVIEW 是一个开发环境（类似的如 Visual Studio 也是一个开发环境），在这个环境下编写的代码就是 G 语言代码（类似的如在 Visual Studio 下写出的C代码）。

https://blog.qizhen.xyz/2007/04/14/g-%E8%AF%AD%E8%A8%80

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Danny
2010年04月7日 00:30 编辑
那算來HP/ Agilent 的 VEE 也是 "G語言" 一族了http://zh.wikipedia.org/zh-tw/%E8%A6%96%E8%A6%BA%E5%8C%96%E7%A8%8B%E5%BC%8F%E8%A8%AD%E8%A8%88%E8%AA%9E%E8%A8%80

回复
Yuanxing
2007年04月25日 09:29 编辑
您好，得到您的指点非常高兴。
我们的平台其实都是ivi驱动，但是不能直接用labview.我想我们新的平台可能会用labview.
谢谢，有什么问题我还会麻烦您的。
从您这儿得到了答案，真好。

回复
善在琳心
2007年04月18日 10:10 编辑
看您的文章受益匪浅，请问可以再写写您对G语言编程思想的理解吗？
我现在也正在用LabVIEW编程，但是很多时候想问题还是文本语言的想法，然后再转成LabVIEW的思想来实现功能的，感觉有些不对！

回复
Qizhen
2007年04月16日 01:11 编辑
把 VI 编译成 DLL，我在 LabVIEW 7 下没有做过这件事情，所以不能肯定。但是 LabVIEW 8.2 我是用过的，多个 VI 可以编译在同一个 DLL 内，每个 VI 为一个被包装成一个外部函数。既然你们买了 8.2，还是用 8.2 吧。
NI DAQ 我也没有太用过，他们提供的驱动主要是针对 LabVIEW 的，在LabVIEW下使用会比在其他语言下使用方便的多。你们的整个系统软件为什么不用 LabVIEW 开发呢？
 

回复
Yuanxing
2007年04月15日 22:22 编辑
你好，好不容易发现有关labview的blog,内容很好，我已经将其收藏。：）
我是一个developer, 要写一些测试设备的驱动，但是我对labview还是无从下手。现在主要是一个对labview比较熟悉的同事先编写vi,将其编译成dll, 然后我再将它import到我的driver里，但是第一，labview编译成de动态库到了我的程序里就成了unmanaged代码，而且，这个dll居然只能包含一个成员函数，请问，这是真的吗？
我们已经采购了labview8.20，但是还没有用它，现在只用7.1，我真想自己编写labview的vi, 这样就能把握整个driver的质量。
第二，我们正在开发ni daq 卡的driver, 如果我们用daq的api开发，是不是自由度更大一些。
我说了这些，不知道您能不能听懂我说什么，其实，我对labview有先天的惧怕心里，虽然有软件，但是没有真正弄懂它。

回复
杰
2007年04月15日 21:47 编辑
我感觉，从8.0开始有了project之后，LV才离通用的编程语言更进一步。呵呵。
LV做递归确实复杂了一点，而且错误难以捕获。
最近一直在看清华的那本LV的高级编程，收获不小。
楼主的本篇Blog，稍显深度不够，嘿嘿。

回复
